### ISSUE TITLE
Translate returning invalid value

### ISSUE THREAD

**Issue by tariellaelius** [ryandrewjohnson/react-localize-redux#66]
After updating to the latest version and switching to `<Translate>` from `localize()`, the following error logs to the console and the app doesn't render at all: 
`Invariant Violation: Translate.render(): A valid React element (or null) must be returned.`
It happens when using `<Translate>` with the `id` property, but if I supply it a render prop function, everything works well. The code is pretty simple: 
```
const Introduction = () => {
  return (
    <section className={style.Introduction}>
      <article className={style.About}>
        <Translate id="introduction" />
      </article>
    </section>
  );
};

export default Introduction;
```

**Comment by ryandrewjohnson** [comment]
Looks like a typings error ... are you using Flow or TypeScript?

**Comment by RikuVan** [comment]
I am using typescript and I just installed version v2.17.3 (from 2.16). I get Translate does not exist when trying to import it (other functions seem to be there if I `import * as T from 'react-localize-redux`).

**Comment by ryandrewjohnson** [comment]
@RikuVan see https://github.com/ryandrewjohnson/react-localize-redux/issues/68#issuecomment-380125253

**Comment by tariellaelius** [comment]
I have just updated to version 2.17.4 and the problem didn't go away. And my project is a JS project, not TypeScript. I've started to suspect it might be some version incompatibility between different packages, as I updated only `react-localize-redux` and not other packages. But then again, why would it work with a render prop function and not simply by `id` reference?

**Comment by ryandrewjohnson** [comment]
Can you provide a copy of your package.json? That way I can try with the same dependencies you're using.

**Comment by ryandrewjohnson** [comment]
Can I see the translation data for this `id="introduction"`?

**Comment by tariellaelius** [comment]
```
{
    "introduction": 
    [
        "dummy text",
        "yoxlama mətn",
        "проверочный текст"        
    ]
}
```
Here is the data, just text replaced with dummy values to make it shorter. I'm not sure about what to post as a working example. I followed the documentation when doing all the other steps, which is probably confirmed by `localize()` and render prop functions working as expected. Still, if I can provide anything specific, let me know.

**Comment by ryandrewjohnson** [comment]
I will need the exact translation data, not dummy values, as I'm suspecting the content could be the issue. I want to use it on my end to see if I can reproduce.

The difference between the render prop, and the passing id method is that copy will be run through `ReactDOMServer.renderToStaticMarkup`, which I'm suspecting may be having difficulties with the data being provided. This method does not get used by the `translate` function which is used by the `render props`.

**Comment by tariellaelius** [comment]
Here is the actual data: 
```
{
    "introduction": 
    [
        " is a fast, reliable and eco-friendly courier service that you can trust!",
        " - əmanətlərinizi etibar edə biləcəyiniz güvənli, sürətli və ekoloji təmiz kuryer xidmətidir.",
        " - надежная, быстрая и экологически чистая курьерская служба, которой можно доверять!"        
    ]
}
```
But I did test with the dummy values, and it still gave me the error.

**Comment by ryandrewjohnson** [comment]
K tested with that data and working fine on my end. Unfortunately there isn't much I can do if I can't reproduce. The only other thing I can suggest is if you can get a version running at https://codesandbox.io/ that is producing the same error, and then share that.

**Comment by tariellaelius** [comment]
So I analyzed the source and located [the line](https://github.com/ryandrewjohnson/react-localize-redux/blob/70353b9ab97081b68a9bd2f49b5dbd89d00b3d9e/src/Translate.js#L122) that is causing the issue. React throws error because it is given a simple string to render as a component. I got it to work by replacing the corresponding line in the locally installed module with `React.createElement('div', {}, translateFn(id, data, options))`, but I'm not sure that it is the best solution and I wonder why it happens only in my project.

**Comment by tariellaelius** [comment]
I also discovered that if the translation data is an HTML string, like `"<div>some text</div>", then no error is thrown, app renders OK, but the elements described in the string are rendered within a `<span>`, though the elements themselves may be block-level elements, like `<div>` in this case, which goes against HTML best practices.

**Comment by ryandrewjohnson** [comment]
As mentioned previously I really need to see a working example that is throwing the error to help out. Right now I don’t have a full view of your source. 

Another option if you don’t want to provide your source is to fork this repo and create a unit test that reproduces the error, and then submit a PR. The Translate component has plenty of unit tests that you can look at for reference. 

Finally, the translation data you provided me worked fine on my end, which is why I believe it’s either a implementation issue or maybe it’s the older version of React. 

I will try and test on my end with older React version to at least rule that out.

**Comment by tariellaelius** [comment]
I was able to reproduce the problem - [![Edit q4zq379o26](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/q4zq379o26) - and indeed, it happens with old version of React, `15.5.4` specifically. Updating React to the latest version fixes the problem. Still, digging deeper into the source, I found that [getLocalizedElement](https://github.com/ryandrewjohnson/react-localize-redux/blob/70353b9ab97081b68a9bd2f49b5dbd89d00b3d9e/src/utils.js#L19) returns either a React element, or just a translation string, depending on a couple of conditions. I may be wrong, but is this function not supposed to return a React element (or maybe null in certain cases), no matter what the conditions?

**Comment by ryandrewjohnson** [comment]
Thanks for taking the time to put this together. I will take a closer look at this when I get some time, but my guess it has to do with React 16 supporting [fragments and strings](https://reactjs.org/blog/2017/09/26/react-v16.0.html#new-render-return-types-fragments-and-strings), while older versions of React do not. I'm guessing there is a scenario in the `Translate` component that results in one of these fragments that aren't supported by React 15 being returned.

I will need to see if there is a clean fix I can do to make things backwards compatible. If not I may just bump the React peer dependency to 16 - meaning users on older versions of React would have to stick to the older API without the `Translate` component.

**Comment by ryandrewjohnson** [comment]
I have just published `v2.17.5` where I have update the peerDependency in the package.json to `"react": "^16.0.0"`, which will now warn user's when installing an older version of React.

I have also updated the documentation with callouts for those using older versions of React along with a new entry in the FAQ page [Can I use Translate component with React 15.x.x?](https://ryandrewjohnson.github.io/react-localize-redux/faq/#can-i-use-translate-component-with-react-15xx).

Thanks again for your help on tracking this one down.
